File: educative/TwoPointers/PairSum.js
const PairSum = (arr, target) => {
	let left = 0
	let right = arr.length - 1

	while (left < right) {
		const sum = arr[left] + arr[right]
		if (sum === target) return [left, right]
		else if (sum < target) left++
		else right--
	}
	return false
}

console.log(PairSum([1, 2, 3, 4, 6], 6))
console.log(PairSum([2, 5, 9, 11], 11))

/*
TIME: O(N) - iterate through all elements
SPACE: O(1) - variables don't scale
*/

File: educative/TwoPointers/PairSum.py
def PairSum(arr, target):
    left = 0;
    right = len(arr) - 1

    while left < right:
        sum = arr[left] + arr[right]
        if sum == target:
            return [left, right]
        elif sum < target:
            left += 1
        else:
            right -= 1
    return false

print(PairSum([1,2,3,4,6], 6))
print(PairSum([2, 5, 9, 11], 11))